src/components/RegistPage.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from "react";
+import { registType } from "../type/type";
+import { apiUrl } from "../service/authService";
+import axios from "axios";
+export default function RegistPage() {
+  const [registData, setRegistData] = useState<registType>({
+    email: "",
+    pwd: "",
+    nickname: "",
+    userName: "",
+    authority: "ROLE_ADMIN",
+    userInfo: "테스트 유저입니다.",
+  });
+
+  const data: registType = {
+    email: registData.email,
+    pwd: registData.pwd,
+    nickname: registData.nickname,
+    userName: registData.userName,
+    authority: registData.authority,
+    userInfo: registData.userInfo,
+  };
+
+  const registHandleSubmit = async (e: { preventDefault: () => void }) => {
+    e.preventDefault();
+    if (registData.pwd.length > 0 && registData.pwd.length < 12)
+      await axios
+        .post(`${apiUrl}`, data)
+        .then((res) => {
+          console.log("회원가입 성공");
+          console.log(res);
+        })
+
+        .catch((error) => {
+          console.log("회원가입 실패");
+          console.log(error);
+        });
+    else alert("비밀번호를 입력해주세요");
```

Comment:
비밀번호만 검증하면 회원가입이 완료되는건가요? 다른 항목들도 검증 로직이 필요할 것 같아요.
현재 해당코드로 봤을 땐 비밀번호와 다른 항목이 같이 비어있을 때, 비밀번호를 입력 해달라고만 오류가 나올 것 같습니다.
그리고 else를 쓰지 않는게 코드를 보는데 조금 더 편리한 것 같아요. depth가 적어져서 코드 보기가 편하고 위에서 if - return구조로 작성하게 되면 위에서 잘못된 조건이 걸렸을 때 return 해버리기 때문에 오류가 발생했을 때 밑의 로직까지 확인할 필요가 없고 코드 파악이 쉬워진다고 생각됩니다 ㅎㅎ https://velog.io/@gomjellie/else-%EC%93%B0%EC%A7%80%EB%A7%88 참고하시면 좋을 것 같아요. 이런식으로 바꿔보면 어떨까요?

```suggestion

   if (registData.pwd.length == 0) {
     alert("비밀번호를 입력해주세요");
     return;
   }
    if (registData.email.length == 0) {
      alert("이메일을 입력해주세요.");
      return;
    }
      
    await axios
        .post(`${apiUrl}`, data)
        .then((res) => {
          console.log("회원가입 성공");
          console.log(res);
        })

        .catch((error) => {
          console.log("회원가입 실패");
          console.log(error);
        });  
```